### PR TITLE
Add FrameState

### PR DESCRIPTION
**Pull request by cbrewster**
r? @asajeffrey 
Might as well get this in now, since it may be a while before I can figure out what is causing the panics on the pending pipelines PR.

---

- [X] `./mach build -d` does not report any errors
- [X] `./mach test-tidy` does not report any errors
- [ ] These changes fix #__ (github issue number if applicable).

- [ ] There are tests for these changes OR
- [X] These changes do not require tests because refactoring.

---

This change is [Reviewable](https://reviewable.io/reviews/servo/servo/12909)

### CODE REVIEW

**Comment by asajeffrey** [comment]
LGTM. @bors-servo: r+

---

Reviewed 1 of 1 files at r1.
Review status: all files reviewed at latest revision, all discussions resolved.

---



*Comments from [Reviewable](https://reviewable.io:443/reviews/servo/servo/12909#-:-KPP9DRzZeGkD6Z7TcOH:buqbv3i)*

File: components/constellation/constellation.rs [hunk 1]
```diff
@@ -219,30 +219,47 @@ pub struct InitialConstellationState {
...
+}
+
 /// Stores the navigation context for a single frame in the frame tree.
 struct Frame {
     id: FrameId,
-    prev: Vec<(PipelineId, Instant)>,
-    current: (PipelineId, Instant),
-    next: Vec<(PipelineId, Instant)>,
+    prev: Vec<FrameState>,
+    current: FrameState,
```
Comment (emilio) [inline components/constellation/constellation.rs]: hmm... what's the point of having `frame.id` now? Isn't it equivalent to `frame.current.id`?

Maybe worth trimming that in a followup.

Comment (ConnorGBrewster) [inline components/constellation/constellation.rs]: frame.id is the FrameId and frame.current.id is a PipelineId

Comment (emilio) [inline components/constellation/constellation.rs]: Whoops, right, sorry, was too late yesterday when I wrote this :-)

**Pull request merged**